src/system_reader.rs

```diff
@@ -699,6 +826,241 @@ mod tests {
         Ok(())
     }
 
+    #[test]
+    fn shared_symbol_table() -> IonResult<()> {
+        let mut map_catalog = MapCatalog::new();
+        map_catalog.put_table(SharedSymbolTable::new(
+            "shared_table".to_string(),
+            1,
+            vec![Some("foo".to_string())],
+        )?);
+        // The stream contains a local symbol table that is not an append.
+        let mut reader = system_reader_with_catalog_for(
+            r#"
+            $ion_symbol_table::{
+                imports: [ { name:"shared_table", version: 1 }],
+            }
+            $10 // "foo"
+          "#,
+            Box::new(map_catalog),
+        );
+        // We step over the LST...
+        assert_eq!(reader.next()?, SymbolTableValue(IonType::Struct));
+        // ...but expect all of the symbols we encounter after it to be in the symbol table,
+        // indicating that the SystemReader processed the LST even though we skipped it with `next()`
+        assert_eq!(reader.next()?, Value(IonType::Symbol));
+        assert_eq!(reader.read_symbol()?, "foo");
+        Ok(())
+    }
+
+    #[test]
+    fn shared_symbol_table_import_with_max_id() -> IonResult<()> {
+        let mut map_catalog = MapCatalog::new();
+        map_catalog.put_table(SharedSymbolTable::new(
+            "shared_table".to_string(),
+            1,
+            vec![Some("foo".to_string())],
+        )?);
+        // The stream contains a local symbol table that is not an append.
+        let mut reader = system_reader_with_catalog_for(
+            r#"
+            $ion_symbol_table::{
+                imports: [ { name:"shared_table", version: 1, max_id: 1 }],
```

Comment:
I think this `max_id` isn't doing anything special (that is: triggering new behavior) since it matches the number of symbols in the table, right? If that's true, could you add a comment saying that?